src/main/java/com/es/phoneshop/web/DemoDataServletContextListener.java

```diff
@@ -0,0 +1,66 @@
+package com.es.phoneshop.web;
+
+import com.es.phoneshop.model.product.ArrayListProductDao;
+import com.es.phoneshop.model.product.Product;
+import com.es.phoneshop.model.product.ProductDao;
+
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Currency;
+import java.util.List;
+import java.util.Random;
+
+public class DemoDataServletContextListener implements ServletContextListener {
+    private ProductDao productDao;
+
+    public DemoDataServletContextListener() {
+        productDao = ArrayListProductDao.getInstance();
+    }
+
+    @Override
+    public void contextInitialized(ServletContextEvent servletContextEvent) {
+        boolean insertDemoData = Boolean.valueOf(servletContextEvent.getServletContext().getInitParameter("insertDemoData"));
+        if (insertDemoData) {
+            getSampleProducts().stream().forEach(product -> productDao.save(product));
+            new Thread(() -> {
```

Comment:
Don't like the way you did it. You can  create some method in DAO to add price history, for example. Also just hint, no need to refactor